File: 240419/밭의 높이를 고르게하기/equalizing-the-height-of-the-field.py
N, H, T = map(int, input().split())
heights = list(map(int, input().split()))
test_heights = heights[:]

cost = 0
min_cost = 100000
for i in range(N):
    if (i + T) <= N:
        for j in range(i, i + T):
            if heights[j] < H:
                while heights[j] != H:
                    heights[j] += 1
                    cost += 1
            elif heights[j] > H:
                while heights[j] != H:
                    heights[j] -= 1
                    cost += 1
        
        min_cost = min(min_cost, cost)
    
    cost = 0
    heights = test_heights[:]

print(min_cost)

File: 260222/밭의 높이를 고르게하기/equalizing-the-height-of-the-field.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int N, H, T;
    cin >> N >> H >> T;

    vector<int> heights(N);
    for (int i = 0; i < N; i++) cin >> heights[i];

    // Please write your code here.
    const int INF = 100000;
    int min_cost = INF;

    for (int i = 0; i + T <= N; i++) 
    {
        long long cost = 0;
        for (int j = i; j < i + T; j++) 
        {
            cost += llabs((long long)heights[j] - H); 
        }
        min_cost = min<long long>(min_cost, cost);
    }

    cout << min_cost << "\n";
    return 0;
}